contracts/GelatoRelayContext.sol

```diff
@@ -67,6 +67,44 @@ abstract contract GelatoRelayContext is GelatoRelayBase {
         _getFeeToken().transfer(_getFeeCollector(), fee);
     }
 
+    // DANGER! Only use with onlyGelatoRelay `_isGelatoRelay` before transferring
+    function _transferFromRelayFee(address _from) internal {
+        _getFeeToken().transferFrom(_from, _getFeeCollector(), _getFee());
+    }
+
+    // DANGER! Only use with onlyGelatoRelay `_isGelatoRelay` before transferring
+    function _transferFromRelayFeeCapped(address _from, uint256 _maxFee)
+        internal
+    {
+        uint256 fee = _getFee();
+        require(
+            fee <= _maxFee,
+            "GelatoRelayContext._transferFromRelayFeeCapped: maxFee"
+        );
+        _getFeeToken().transferFrom(_from, _getFeeCollector(), fee);
+    }
+
+    // DANGER! Only use with onlyGelatoRelay `_isGelatoRelay` before transferring
+    function _transferFromRelayFeeCappedWithPermit(
+        address _from,
+        uint256 _maxFee,
+        uint256 _deadline,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+    ) internal {
+        uint256 fee = _getFee();
+        require(
+            fee <= _maxFee,
+            "GelatoRelayContext._transferFromRelayFeeCappedWithPermit: maxFee"
+        );
+
+        address token = _getFeeToken();
+        token.permit(_from, address(this), _maxFee, _deadline, _v, _r, _s);
```

Comment:
Here we should only `permit` the `fee` not the `maxFee` as otherwise there might remain a lot of unspent allowance thereafter that could even be stolen by other relay users.

Reply:
As discussed in Slack - we simply have to assume Consumer contracts are implemented securely here by devs - making sure that no unspent allowances from these Relay payments via transferFrom can be abused by attackers. 

@bennoprice please add documentation here in the code succinctly explaining that using these methods brings along approvals and potentially unspent allowances, which has to be taken into account by devs using these methods in their contracts. They need to make sure there arent any ways to abuse these allowances.